pvlib/iotools/tmy.py

```diff
@@ -161,33 +158,17 @@ def read_tmy3(filename=None, coerce_year=None, recolumn=True):
        Update: Users Manual. 472 pp.; NREL Report No. TP-581-41364.
     '''
 
-    if filename is None:
-        try:
-            filename = _interactive_load()
-        except ImportError:
-            raise ImportError('Interactive load failed. tkinter not supported '
-                              'on this system. Try installing X-Quartz and '
-                              'reloading')
-
     head = ['USAF', 'Name', 'State', 'TZ', 'latitude', 'longitude', 'altitude']
 
-    if str(filename).startswith('http'):
-        request = Request(filename, headers={'User-Agent': (
-            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) '
-            'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 '
-            'Safari/537.36')})
-        response = urlopen(request)
-        csvdata = io.StringIO(response.read().decode(errors='ignore'))
-    else:
-        # assume it's accessible via the file system
-        csvdata = open(str(filename), 'r')
-
-    # read in file metadata, advance buffer to second line
-    firstline = csvdata.readline()
-    if 'Request Rejected' in firstline:
-        raise IOError('Remote server rejected TMY file request')
-
-    meta = dict(zip(head, firstline.rstrip('\n').split(",")))
+    with open(filename, 'r') as csvdata:
+        # read in file metadata, advance buffer to second line
+        firstline = csvdata.readline()
+        meta = dict(zip(head, firstline.rstrip('\n').split(",")))
```

Comment:
This line doesn't need to be in the `with` block, but it seems clearer to me to leave it here next to the definition of `firstline`.

Reply:
here or above the dict assignment is fine with me. I think the nuance about the buffer is more clear if this line is moved outside the block but up to you.

Reply:
Agreed.  Moved outside the block.